### DISCUSSION TITLE
post/multiple-nuget-single-repo

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#10] [Announcements]
# post/multiple-nuget-single-repo

Mark Heath's Development Blog

https://markheath.net/post/multiple-nuget-single-repo

**Comment by koljada** [comment]
I don't think this is accurate. I believe you have both packages created only because you executed `dotnet pack` command for a whole solution. But if you run it against your B project(considering B project has its own directory) you should have only single LibraryB project with LibraryA package not created/updated. I guess `dotnet pack` cannot handle dependencies, it just creates packages for everything it can reach

**Comment by markheath** [comment]
Sure, with this setup I do `dotnet pack` for the whole solution.

**Comment by DanCarterGMSL** [comment]
Thank you so much for posting this article! Like you, I had assumed that this would be really difficult to do and was surprised that NuGet was able to do what I wanted by default.

Even when things are easy it's so useful to have this article as it solved a problem I had (wrongly) assumed would be hard and so was putting off solving ...

**Comment by prmeyn** [comment]
Thanks for the article, I'm trying the same with Github actions to build all dependencies into a single nuget package.

So based on your Example:  `LibraryA` in my case is `GeoIpServices.csproj`
and I have 2 `LibraryB's` `GeoIpCommon.csproj` & `GeoIpStack.csproj`

You can see them referenced here: https://github.com/prmeyn/GeoIpServices/blob/main/GeoIpServices/GeoIpServices.csproj

The YAML file to build the package and deploy to nuget.org only builds `LibraryA` without the dependencies: https://github.com/prmeyn/GeoIpServices/blob/main/.github/workflows/release.yml

You can see it here: https://nuget.info/packages/GeoIpServices/2.0.4

Any ideas how I could get all the dll files in a single nuget package without having to make a separate nuget for each? I really don't want to merge the code into a single class library.
